gcsstore.go

```diff
@@ -207,6 +193,34 @@ func (g *GcsFS) Delete(obj string) error {
 	return nil
 }
 
+type GcsObjectIterator struct {
+	g    *GcsFS
+	iter *storage.ObjectIterator
+}
+
+func (it *GcsObjectIterator) Next() (Object, error) {
+	var lasterr error = nil
+	retryCt := 0
+
+	for {
+		o, err := it.iter.Next()
+		if err == nil {
+			return newObjectFromGcs(it.g, o), nil
+		} else if err == iterator.Done {
+			return nil, err
+		}
+		lasterr = err
+		if retryCt < 5 {
```

Comment:
1. at LN 133 we should memorize the ctx passed in.
2. here we should check if the it.ctx.Done() has been closed.